### ISSUE TITLE
Perform Vault operations concurrently

### ISSUE THREAD

**Issue by mattlqx** [fishi0x01/vsh#78]
Presently all Vault operations are serial, even for reads/writes that operate recursively on paths. These paths are determined by list api calls to find path targets, but after that a for loop iterates over those paths to get or put the data. We could utilize Go channels for this portion to run the read/write api calls simultaneously.

**Comment by fishi0x01** [comment]
You are right. Parallelizing the calls could save quite some time when recursively operating on a decent amount of secrets in a path. 

I like your idea :+1: 

I can take a look at this if you want - but since you self-assigned this I assume you already gave it a shot? :sunglasses:

**Comment by fishi0x01** [comment]
Fixed in https://github.com/fishi0x01/vsh/pull/126
